video/out/opengl/hwdec_ios.m

```diff
@@ -0,0 +1,295 @@
+    int i;
+
+    for (i = 0; i < MP_MAX_PLANES; i++) {
+        if (p->gl_planes[i]) {
+            CFRelease(p->gl_planes[i]);
+            p->gl_planes[i] = NULL;
+        }
+    }
+
+    CVOpenGLESTextureCacheFlush(p->gl_texture_cache, 0);
+}
+
+static int map_frame(struct gl_hwdec *hw, struct mp_image *hw_image,
+                     struct gl_hwdec_frame *out_frame)
+{
+    struct priv *p = hw->priv;
+    GL *gl = hw->gl;
+
+    CVPixelBufferRelease(p->pbuf);
+    p->pbuf = (CVPixelBufferRef)hw_image->planes[3];
+    CVPixelBufferRetain(p->pbuf);
+
+    uint32_t cvpixfmt = CVPixelBufferGetPixelFormatType(p->pbuf);
+    struct vt_format *f = vt_get_gl_format(cvpixfmt);
+    if (!f) {
+        MP_ERR(hw, "CVPixelBuffer has unsupported format type\n");
+        return -1;
+    }
+
+    const bool planar = CVPixelBufferIsPlanar(p->pbuf);
+    const int planes  = CVPixelBufferGetPlaneCount(p->pbuf);
+    assert(planar && planes == f->planes || f->planes == 1);
+
+    cleanup_textures(hw);
+
+    for (int i = 0; i < f->planes; i++) {
+        gl->ActiveTexture(GL_TEXTURE0 + i);
+
+        CVReturn err = CVOpenGLESTextureCacheCreateTextureFromImage(
+            kCFAllocatorDefault,
+            p->gl_texture_cache,
+            p->pbuf,
+            NULL,
+            GL_TEXTURE_2D,
+            f->gl[i].gl_internal_format,
+            CVPixelBufferGetWidthOfPlane(p->pbuf, i),
+            CVPixelBufferGetHeightOfPlane(p->pbuf, i),
+            f->gl[i].gl_format,
+            f->gl[i].gl_type,
+            i,
+            &p->gl_planes[i]);
+
+        if (err != noErr)
+            MP_ERR(hw, "error creating texture for plane %d: %d\n", i, err);
+
+        gl->BindTexture(CVOpenGLESTextureGetTarget(p->gl_planes[i]),
+                        CVOpenGLESTextureGetName(p->gl_planes[i]));
+        gl->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+        gl->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+        gl->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+        gl->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+
+        out_frame->planes[i] = (struct gl_hwdec_plane){
+            .gl_texture = CVOpenGLESTextureGetName(p->gl_planes[i]),
+            .gl_target = CVOpenGLESTextureGetTarget(p->gl_planes[i]),
+            .tex_w = CVPixelBufferGetWidthOfPlane(p->pbuf, i),
+            .tex_h = CVPixelBufferGetHeightOfPlane(p->pbuf, i),
+        };
+        snprintf(out_frame->planes[i].swizzle, sizeof(out_frame->planes[i].swizzle),
+                 "%s", f->gl[i].swizzle);
+    }
+
+    return 0;
+}
+
+static void destroy(struct gl_hwdec *hw)
+{
+    struct priv *p = hw->priv;
+
+    cleanup_textures(hw);
+
+    CVPixelBufferRelease(p->pbuf);
+    CFRelease(p->gl_texture_cache);
+    p->gl_texture_cache = NULL;
+
+    hwdec_devices_remove(hw->devs, &p->hwctx);
+}
+
+const struct gl_hwdec_driver gl_hwdec_videotoolbox = {
+    .name = "videotoolbox",
```

Comment:
The way it's currently set up, either osx or ios backend is built, so both are called videotoolbox. This seems preferable to me, as it should not be possible to compile both implementations together anyway.